### PR TITLE
Added backup destination types for Tencent COS, AWS S3 and S3 compatible services

### LINKED ISSUE #521: [Feature]: Support backup to S3
### Describe your use case

Support backup to S3

### Describe the solution you'd like

Support backup to S3

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by powerfooI**
This PR will close #521

COS, AWS s3, Huawei OBS and GCP Storage are all tested.

### CODE REVIEW

File: api/v1alpha1/obtenantbackuppolicy_webhook.go [hunk 1]
```diff
@@ -387,3 +292,57 @@ func validateScheduleFormat(schedule string, fldPath *field.Path) *field.Error {
...
+	if !pattern.MatchString(dest.Path) {
+		return field.Invalid(field.NewPath("spec").Child(fieldName).Child("destination"), dest.Path, "invalid backup destination path, the path format should be "+pattern.String())
+	}
+	if dest.Type != constants.BackupDestTypeNFS && dest.OSSAccessSecret == "" {
+		return field.Invalid(field.NewPath("spec").Child(fieldName).Child("destination"), dest.OSSAccessSecret, "OSSAccessSecret is required when backing up data to OSS, COS or S3")
+	}
+	secret := &v1.Secret{}
+	err := bakClt.Get(context.Background(), types.NamespacedName{
+		Namespace: r.GetNamespace(),
+		Name:      dest.OSSAccessSecret,
```
Comment (chris-sun-star) [inline api/v1alpha1/obtenantbackuppolicy_webhook.go:313]: is it necessary to change the definition to AccessSecret and make OSSAccessSecret deprecated

File: api/constants/backup.go [hunk 2]
```diff
@@ -45,8 +49,11 @@ const (
 )
 
 const (
-	BackupDestTypeOSS types.BackupDestType = "OSS"
-	BackupDestTypeNFS types.BackupDestType = "NFS"
+	BackupDestTypeOSS          types.BackupDestType = "OSS"
+	BackupDestTypeNFS          types.BackupDestType = "NFS"
+	BackupDestTypeCOS          types.BackupDestType = "COS"
+	BackupDestTypeS3           types.BackupDestType = "S3"
+	BackupDestTypeS3Compatible types.BackupDestType = "S3_COMPATIBLE"
```
Comment (chris-sun-star) [inline api/constants/backup.go:56]: I don't think it's necessary to add S3_COMPATIBLE, just use  S3 if it's using S3 protocol

Comment (powerfooI) [reply api/constants/backup.go:56]: But the parameters they two need is not the same.

File: api/v1alpha1/obtenantbackuppolicy_webhook.go (hunk 1 above)
Comment (powerfooI) [reply api/v1alpha1/obtenantbackuppolicy_webhook.go:313]: I think OSS is short for Object Storage Service which is suitable for all object storage providers.

Comment (chris-sun-star) [reply api/v1alpha1/obtenantbackuppolicy_webhook.go:313]: OK

File: api/constants/backup.go (hunk 2 above)
Comment (chris-sun-star) [reply api/constants/backup.go:56]: OK, found it

Comment (powerfooI) [reply api/constants/backup.go:56]: `s3Region` field is required in destination of `S3` type while it's not required in `S3Compatible` type.

**Review by chris-sun-star — APPROVED** [review]
LGTM

**Pull request merged**